srv/modules/runners/orderednodes.py

```diff
@@ -0,0 +1,38 @@
+import salt.client
+
+
+def _preserve_order_sorted(seq):
+    """
+    Getting rid of duplicates in python could be solved by
+    casting a list() to a set() and back to a list()
+    `list(set(list_in_question))`
+    This method will mess with the sorting though.
+    As we rely on the sorting in this scenario, we have to use this
+    helper.
+    """
+    seen = set()
+    return [x for x in seq if x not in seen and not seen.add(x)]
+
+
+def unique(cluster=None, exclude=[]):
+    """ 
+    Assembling a list of nodes.
+    Ordered(MON < OSD < MDS < RGW < IGW)  
```

Comment:
That's the suggested order of upgrading nodes/services

Arrows would've been less confusing and more descriptive